File: planetiler-custommap/src/main/java/com/onthegomap/planetiler/custommap/Power.java
import com.onthegomap.planetiler.FeatureCollector;
import com.onthegomap.planetiler.Planetiler;
import com.onthegomap.planetiler.Profile;
import com.onthegomap.planetiler.config.Arguments;
import com.onthegomap.planetiler.reader.SourceFeature;
import java.nio.file.Path;

class Power implements Profile {
  public void processFeature(SourceFeature source, FeatureCollector features) {
    if (source.canBeLine() && source.hasTag("power", "line")) {
      features
        .line("power")
        .setMinZoom(7)
        .inheritAttrFromSource("power")
        .inheritAttrFromSource("voltage")
        .inheritAttrFromSource("cables")
        .inheritAttrFromSource("operator");
    } else if (source.isPoint() && source.hasTag("power", "pole")) {
      features
        .point("power")
        .setMinZoom(13)
        .inheritAttrFromSource("power")
        .inheritAttrFromSource("ref")
        .inheritAttrFromSource("height")
        .inheritAttrFromSource("operator");
    }

  }

  public String name() {
    return "power";
  }

  void main(String[] argv) throws Exception {
    Arguments args = Arguments.fromArgsOrConfigFile(argv);
    var area = args.getString("area", "geofabrik area to download", "massachusetts");
    Planetiler.create(args)
      .setProfile(new Power())
      .addOsmSource("osm", Path.of("data/sources/" + area + ".osm.pbf"), "geofabrik:" + area)
      .overwriteOutput(Path.of("data/buildings.pmtiles"))
      .run();
  }
}

File: planetiler-experimental/src/test/resources/power.lua
-- Example lua profile that emits power lines and poles from an openstreetmap source
-- useful for hot air ballooning

-- The planetiler object defined in LuaEnvironment.PlanetilerNamespace is the interface for sharing
-- data between lua scripts and Java
planetiler.output.name = "Power"
planetiler.output.description = "Simple"
planetiler.output.attribution =
'<a href="https://www.openstreetmap.org/copyright" target="_blank">&copy;OpenStreetMap contributors</a>'
planetiler.examples = "power.spec.yaml"
planetiler.output.path = { "data", "power.pmtiles" }

local area = planetiler.args:get_string("area", "geofabrik area to download", "massachusetts")

planetiler.add_source('osm', {
  type = 'osm',
  url = 'geofabrik:' .. area,
  -- any java method or field that takes a Path can be called with a list of path parts from lua
  path = { 'data', 'sources', area .. '.osm.pbf' }
})

function planetiler.process_feature(source, features)
  if source:can_be_line() and source:has_tag("power", "line") then
    features
        :line("power")
        :set_min_zoom(7)
        :inherit_attr_from_source("power")
        :inherit_attr_from_source("voltage")
        :inherit_attr_from_source("cables")
        :inherit_attr_from_source("operator")
  elseif source:isPoint() and source:has_tag("power", "pole") then
    features
        :point("power")
        :set_min_zoom(13)
        :inherit_attr_from_source("power")
        :inherit_attr_from_source("ref")
        :inherit_attr_from_source("height")
        :inherit_attr_from_source("operator")
  end
end
